### PR TITLE
Use react router for invoice page

### PR DESCRIPTION
**Pull request by apoorv-mishra**
## Notion card
NA
## Summary

This PR moves implements routing for invoice page on FE
## Preview

Screenshot 2022-04-22 at 8 01 38 PM

Fixes ☝️ 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have added automated tests for my code

### CODE REVIEW

File: app/controllers/internal_api/v1/invoices_controller.rb [hunk 1]
```diff
@@ -63,6 +71,20 @@ def invoice
...
+        .find(params[:id]).as_json(include: [:invoice_line_items, :client])
+        .merge(company: {
+          id: current_company.id,
+          logo: current_company.logo.attached? ? polymorphic_url(current_company.logo) : "",
+          name: current_company.name,
+          phone_number: current_company.business_phone,
+          address: current_company.address,
+          country: current_company.country,
+          currency: current_company.base_currency
+        })
```
[the remark this answers is not in the document]
Comment (apoorv-mishra) [reply app/controllers/internal_api/v1/invoices_controller.rb]: Removed altogether and handled in the view layer.

**Review by supriya3105 — APPROVED** [review]

**Review by mohd-anas-ansari — APPROVED** [review]

File: app/views/internal_api/v1/invoices/show.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+json.discount invoice.discount
+json.status invoice.status
+json.client do
+  json.id invoice.client.id
+  json.name invoice.client.name
+  json.email invoice.client.email
+  json.phone invoice.client.phone
+  json.address invoice.client.address
+end
+json.company do
```
Comment (shivamsinghchahar) [inline app/views/internal_api/v1/invoices/show.json.jbuilder:23]: Move client, company and invoice_line_items to partials

File: app/views/internal_api/v1/invoices/show.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  json.id invoice.client.company.id
+  json.logo invoice.client.company.logo.attached? ? polymorphic_url(invoice.client.company.logo) : ""
+  json.name invoice.client.company.name
+  json.phone_number invoice.client.company.business_phone
+  json.address invoice.client.company.address
+  json.country invoice.client.company.country
+  json.currency invoice.client.company.base_currency
+end
+json.invoice_line_items invoice.invoice_line_items do |invoice_line_item|
+  json.id invoice_line_item.id
```
Comment (shivamsinghchahar) [inline app/views/internal_api/v1/invoices/show.json.jbuilder:33]: `N+1` queries here, need to `includes(:invoice_line_items)` while fetching invoice from DB in `invoice` method.

**Review by shivamsinghchahar — CHANGES_REQUESTED** [review]
@apoorv-mishra Please look at the comments below

File: app/views/internal_api/v1/invoices/show.json.jbuilder (hunk 2 above)
Comment (apoorv-mishra) [reply app/views/internal_api/v1/invoices/show.json.jbuilder]: Fixed.

File: app/views/internal_api/v1/invoices/show.json.jbuilder (hunk 3 above)
Comment (apoorv-mishra) [reply app/views/internal_api/v1/invoices/show.json.jbuilder]: Fixed.

**Comment by supriya3105** [comment]
@shivamsinghchahar Apoorv has incorporated the feedback comments. Have a look and approve if the PR is good to be merged.

File: app/javascript/src/components/Invoices/Invoice/index.tsx [hunk 4]
```diff
@@ -1,16 +1,46 @@
...
-    </div>
-  </>
-);
+const Invoice = () => {
+  const params = useParams();
+
+  const [status, setStatus] = React.useState<InvoiceStatus>(
+    InvoiceStatus.IDLE
+  );
+  const [invoice, setInvoice] = useState<any>();
```
Comment (shivamsinghchahar) [inline app/javascript/src/components/Invoices/Invoice/index.tsx:15]: Just one tiny change, the initial value should be `null`.
```suggestion
  const [invoice, setInvoice] = useState<any>(null);
```

**Review by shivamsinghchahar — APPROVED** [review]
Just one small change, rest LGTM 🚀

File: app/javascript/src/components/Invoices/Invoice/index.tsx (hunk 4 above)
Comment (apoorv-mishra) [reply app/javascript/src/components/Invoices/Invoice/index.tsx]: Fixed.

**Pull request merged**